### PR TITLE
Stat/Gauge/BarGauge: Shows default cursor when missing links

### LINKED ISSUE #24105: Stat/Gauge/BarGauge has context menu cursor even when there are no data links
Stat/Gauge/BarGauge has context menu cursor even when there are no data links

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Same as https://github.com/grafana/grafana/pull/24232 but with a slightly fresh approach.

**Which issue(s) this PR fixes**:
Closes #24105

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx [hunk 1]
```diff
@@ -6,11 +6,12 @@ import { css } from 'emotion';
...
 interface DataLinksContextMenuProps {
   children: (props: { openMenu?: React.MouseEventHandler<HTMLElement>; targetClassName?: string }) => JSX.Element;
+  hasLinks: boolean;
   links?: () => LinkModel[];
 }
 
-export const DataLinksContextMenu: React.FC<DataLinksContextMenuProps> = ({ children, links }) => {
-  if (!links) {
+export const DataLinksContextMenu: React.FC<DataLinksContextMenuProps> = ({ children, links, hasLinks }) => {
+  if (!links || !hasLinks) {
```
Comment (dprokop) [inline packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx:14]: Isn't !hasLinks enough?

File: public/app/plugins/panel/bargauge/BarGaugePanel.tsx [hunk 2]
```diff
@@ -1,27 +1,24 @@
...
 export class BarGaugePanel extends PureComponent<PanelProps<BarGaugeOptions>> {
   renderValue = (valueProps: VizRepeaterRenderValueProps<FieldDisplay, DisplayValueAlignmentFactors>): JSX.Element => {
     const { options } = this.props;
     const { value, alignmentFactors, orientation, width, height } = valueProps;
-    const { field, display, view, colIndex } = value;
+    const { field, display, view, colIndex, hasLinks, getLinks } = value;
 
     return (
-      <DataLinksContextMenu links={value.getLinks}>
+      <DataLinksContextMenu links={getLinks} hasLinks={hasLinks}>
```
Comment (dprokop) [inline public/app/plugins/panel/bargauge/BarGaugePanel.tsx:21]: Hm, I'm not a fan of boolean props to decide whether or not something should be rendered. I would go one step further here and probably make DataLinksContextMenu `links` property non optional and only wrap in Context menu if the links are there. WDYT?

**Review by dprokop — APPROVED** [review]
This is also looking good, think I like this better with `hasLinks` in the FieldDisplay. Only one nit I found that can be addressed if you agree @hugohaggmark

File: packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx (hunk 1 above)
Comment (hugohaggmark) [reply packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx:14]: Yes, thank you!

File: public/app/plugins/panel/bargauge/BarGaugePanel.tsx (hunk 2 above)
Comment (hugohaggmark) [reply public/app/plugins/panel/bargauge/BarGaugePanel.tsx:21]: Yes, I started down this path but then that adds another if statement for each component and they grow but again I like the quick exit strategy. So I agree and I will close https://github.com/grafana/grafana/pull/24232

**Comment by hugohaggmark** [comment]
> This is also looking good, think I like this better with `hasLinks` in the FieldDisplay. Only one nit I found that can be addressed if you agree @hugohaggmark

I totally agree, will push changes soon!

**Review by torkelo — APPROVED** [review]
Great job!

**Pull request merged**